src/components/grid.jsx

```diff
@@ -388,6 +422,35 @@ class Grid extends React.Component {
       this.props.highlightedSnake
     );
 
+    if (!this.props.highlightedSnake) {
+      // track all of the grid cells that will have a snake part drawn in them.  Successive snake parts 
+      // drawn in the same cell need to be flagged so they render differently and layer properly
+      let renderedSnakeParts = Array(this.props.rows);
```

Comment:
Calling the `Array` constructor here seems strange considering these are already arrays, same with line 430. I assume there must be a reason that I don't see?

Reply:
From my understanding, `props.rows` and `props.cols` are numbers not arrays.  Effectively, what I need to do in this code is create an initialized 2d array of bools so that I can track when a cell is already "used" by another snake and notify subsequent snake parts that they are overlapping.

Reply:
Ah ok, I haven't pulled this down and tested yet so that is a reasonable explanation.